src/main/java/org/folio/services/consortium/ConsortiumConfigurationService.java

```diff
@@ -3,43 +3,76 @@
 import static org.folio.services.consortium.util.ConsortiumConfigurationFields.CENTRAL_TENANT_ID;
 import static org.folio.services.consortium.util.ConsortiumConfigurationFields.CONSORTIUM_ID;
 import static org.folio.services.consortium.util.ConsortiumConfigurationFields.USER_TENANTS;
+import static org.folio.util.HeaderUtils.prepareHeaderForTenant;
 
+import java.util.Map;
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
 
 import com.github.benmanes.caffeine.cache.AsyncCache;
 import com.github.benmanes.caffeine.cache.Caffeine;
 
+import io.vertx.core.Context;
 import io.vertx.core.Future;
 import io.vertx.core.Vertx;
 import lombok.extern.log4j.Log4j2;
 import org.folio.models.ConsortiumConfiguration;
 import org.folio.rest.core.RestClient;
 import org.folio.rest.core.models.RequestContext;
 import org.folio.rest.core.models.RequestEntry;
+import org.folio.rest.jaxrs.model.Setting;
 import org.folio.rest.tools.utils.TenantTool;
+import org.folio.services.setting.SettingService;
+import org.folio.services.setting.util.SettingKey;
 import org.folio.util.ResourcePath;
 import org.springframework.beans.factory.annotation.Value;
 
 @Log4j2
 public class ConsortiumConfigurationService {
 
   private final RestClient restClient;
+  private final SettingService settingService;
   private final AsyncCache<String, Optional<ConsortiumConfiguration>> asyncCache;
 
   @Value("${orders-storage.cache.consortium-data.expiration.time.seconds:300}")
   private long cacheExpirationTime;
 
-  public ConsortiumConfigurationService(RestClient restClient) {
+  public ConsortiumConfigurationService(RestClient restClient, SettingService settingService) {
     this.restClient = restClient;
+    this.settingService = settingService;
 
     asyncCache = Caffeine.newBuilder()
       .expireAfterWrite(cacheExpirationTime, TimeUnit.SECONDS)
       .executor(task -> Vertx.currentContext().runOnContext(v -> task.run()))
       .buildAsync();
   }
 
+  public Future<String> getCentralTenantId(Context context, Map<String, String> headers) {
+    var requestContext = new RequestContext(context, headers);
+    return getConsortiumConfiguration(requestContext)
+      .compose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isEmpty()) {
+          log.info("getCentralTenantId:: No settings were found");
```

Comment:
we will have this log for each non ecs event processing, can be debug or removed